sequence_task_1/student/include/impl.h

```diff
@@ -8,33 +8,86 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    typedef Song_t value_type;
+    typedef Song_t& reference;
+    typedef const Song_t& const_reference;
```

Comment:
I'd just stick to the container traits, so in this case we'll introduce a requirement for our class template params
So there is no reason to define them manually